### PR TITLE
Adapt `HotSbtBenchmark` to benchmark whole sbt projects

### PR DESCRIPTION
**Pull request by Duhemm**

### CODE REVIEW

**Comment by Duhemm** [comment]
I haven't taken care of the `ScalacBenchmark` yet. Those are more complicated, because you can't let sbt do all the heavy lifting. I think that it's enough to get started.

File: compilation/src/main/scala/scala/tools/nsc/HotSbtBenchmark.scala [hunk 1]
```diff
@@ -15,80 +15,60 @@ import org.openjdk.jmh.annotations._
...
+       |  override def globalSettings: Seq[Setting[_]] = Seq(
+       |    cleanClasses := Def.taskDyn {
+       |      cleanClassesTask.all(ScopeFilter(inAnyProject))
+       |    }.value
+       |  )
        |
-       |$compileRawSetting
-       |
-       |$extraBuild
-       |// TODO support .java sources
```
Comment (jvican) [inline compilation/src/main/scala/scala/tools/nsc/HotSbtBenchmark.scala:78]: Do we support Java sources?

**Review by jvican — APPROVED** [review]
Makes sense, LGTM

File: compilation/src/main/scala/scala/tools/nsc/HotSbtBenchmark.scala (hunk 1 above)
Comment (Duhemm) [reply compilation/src/main/scala/scala/tools/nsc/HotSbtBenchmark.scala:78]: As long as we don't run the `ScalacBenchmark`, yes.

**Pull request merged**